pkg/identity/cache/cache.go

```diff
@@ -201,7 +201,8 @@ func (w *identityWatcher) stop() {
 
 // LookupIdentity looks up the identity by its labels but does not create it.
 // This function will first search through the local cache and fall back to
-// querying the kvstore.
+// querying all connected kvstores (first the main kvstore, followed by any
+// watched remote kvstores in clustermesh)
 func (m *CachingIdentityAllocator) LookupIdentity(ctx context.Context, lbls labels.Labels) *identity.Identity {
```

Comment:
Note to reviewers: While we surely want to fix the lookup in this function for the API to work correctly (any identity returned by `/v1/identities` should also be fetchable via labels`), the only non-API call site (outside of unit tests) of this function is in the IPCache:

https://github.com/cilium/cilium/blob/3a49c6c53dce1a768756f76b1c29e38c65e7d54a/pkg/ipcache/cidr.go#L118

IPCache is performing a lookup on CIDR labels. I don't think we want to send that lookup to remote kvstores (since CIDR identies are always local), so maybe we have to introduce an additional lookup function for the IPCache usecase? Unfortunatley I'm not too familiar with local identity management, so feedback welcome.

Reply:
Yeah, CIDR identities are purely local to the agent (not even shared across the cluster), so we should follow up to change the lookup function to reflect this usage.